### PR TITLE
[ResourceAction] LPS-117531 Read portlet resources per portlet after initializing portlet models

### PR DESCRIPTION
**Pull request by dantewang**

### CODE REVIEW

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
More tests previously tested in https://github.com/dantewang/liferay-portal/pull/1681 and reviewed by Victor.

**Comment by tinatian** [comment]
@dantewang 
This is not what I want, or the logic here is not completed. It is weird that I add "portlet" as an extra parameter to the API method, and pass in null on some cases, we need to separate it into different method.

I would like to see we separate portlet resource reading and module resource reading. But both portlet resource and module resource reading logic are still in the same method:

```
private void _read(
			Portlet portlet, String servletContextName, Document document,
			Set<String> resourceNames)
```

Basically, I want to see we have two read methods:
readPortletResource
readModelResource
And you invoke corresponding method when you want to read either portlet resource or model resource, so that we will make it clear when portlet resource is read, and when model resource is read.

Please separate the logic and resend.

Thanks.
Tina.
